### ISSUE TITLE
Polyfill doesn't handle text elements

### ISSUE THREAD

**Issue by Sora2455** [mozilla/sanitizer-polyfill#141]
On MDN, the example 
```js
const unsanitized_string = "abc <script>alert(1)<" + "/script> def";  // Unsanitized string of HTML
const sanitizer1 = new Sanitizer();  // Default sanitizer;

// Get the Element with id "target" and set it with the sanitized string.
document.getElementById("target").setHTML(unsanitized_string, {sanitizer: sanitizer1});
```
Is expected to leave the target with the target with the innerHTML of "abc  def": but the [demo page](https://mozilla.github.io/sanitizer-polyfill/demo/) shows an empty string.

Likewise, this example from the Sanitizer explainer doc:

```js
const for_display = new Sanitizer({
  allowElements: ['span', 'em', 'strong', 'b', 'i'],
  allowAttributes: {'class': ['*']}
});

const untrusted_example = "Well, <em class=nonchalant onclick='alert(\'General Kenobi\');'><a href='https://obiwan.example/home.php'>hello there<a>!"

element.setHTML(untrusted_example, {sanitizer: sanitizer});
```

Is expected to leave the element with an innerHTML of `Well, <em class="nonchalant"><a href='https://obiwan.example/home.php'>hello there<a>!</em>`, but in the demo page this results in `<em class="nonchalant"><a href="https://obiwan.example/home.php">hello there</a><a>!</a></em>` (missing the "Well, " at the start).

**Comment by mozfreddyb** [comment]
Indeed, we seem to swallow the text node. I will dig further.

**Comment by Sora2455** [comment]
The merged PR does fix this problem in that the text nodes are now present, but the div tag is now cloned inside itself. This isn't so much a problem for a div, but it is for a table or select element (the other two example elements on the demo page).

**Comment by mozfreddyb** [comment]
Yeah, I believe the fix we wanted wasn't `...context.children` but `...context.childNodes`. PR coming.

**Comment by mozfreddyb** [comment]
Works for me. Thanks for the quick feedback @Sora2455!
